dlc2nwb/utils.py

```diff
@@ -37,6 +37,33 @@ def get_movie_timestamps(movie_file, VARIABILITYBOUND=1000):
             "Variability of timestamps suspiciously small. See: https://github.com/DeepLabCut/DLC2NWB/issues/1"
         )
 
+    if any(timestamps[1:] == 0):
+        # Infers times when OpenCV provides 0s
+        warning_msg = "Removing"
+        timestamp_zero_count = np.count_nonzero(timestamps == 0)
+        timestamps[timestamps == 0] = np.nan # replace 0s with nan
+
+        if infer_timestamps:
+            warning_msg = "Replacing"
+            timestamps = _infer_nan_timestamps(timestamps)
+
+        warnings.warn( # warns user of percent of 0 frames
+            "%s cv2 timestamps returned as 0: %f%%"
+            % (warning_msg, ( timestamp_zero_count / len(timestamps) * 100))
+        )
+
+    return timestamps
+
+
+def _infer_nan_timestamps(timestamps):
+    """Given np.array, interpolate nan values using index * sampling rate"""
+    bad_timestamps_mask = np.isnan(timestamps)
+    bad_timestamps_indexes = np.argwhere(bad_timestamps_mask)[:, 0]
+    estimated_sampling_rate = np.mean(np.diff(timestamps[~bad_timestamps_mask]))
```

Comment:
If nans are systematically found at the end, this is valid; if they can be found in between valid time stamps though, `diff` will yield erroneously high time gaps. I don't know how relevant that is, but if you think it may be important for some applications, `bad_timestamps_mask` could be grouped into subsequences of consecutive `True`, and sampling rate estimated based on these.

Reply:
Hi @jeylau - Thanks for your feedback. I don't know with certainty that the bad timestamps will be at the end. I just pushed a commit that finds the runs of good timestamps and uses those diffs to generate an average estimated sampling rate. Let me know what you think.